### ISSUE TITLE
"Dynamic requires" not at all working.

### ISSUE THREAD

**Issue by Qix-** [zeit/ncc#383]
I have a simple wrapper around `require()` but it seems to have broken my deployment.

```javascript
const reaction = name => {
	const res = require(name);
	res.name = name.split('/').pop();
	return res;
};

const reactions = {
	repository: {
		publicized: reaction('../reactions/publicize-repository')
	},
	issues: {
		opened: [reaction('../reactions/check-triage')],
		label: [reaction('../reactions/check-triage')]
	}
};
```

Worked just fine before I added the wrapper. Isn't this what "dynamic require" means in this repository's description?

**Comment by rauchg** [comment]
Dynamic requires that are statically analyzable. We might need to add a *

Not sure if we can improve the static analyzer for this case, though. cc @guybedford

**Comment by guybedford** [comment]
We could add some simple analysis to check for require wrapper functions, certainly, it would just be a little bit of work to fold that into the analysis.

**Comment by rauchg** [comment]
@Qix- we are on it.

**Comment by guybedford** [comment]
This has been implemented in https://github.com/zeit/webpack-asset-relocator-loader/pull/34, although will only land with the next ncc major.

**Comment by aequasi** [comment]
I think this might solve: https://github.com/zeit/now-builders/issues/508

Is there any way to test this with `now dev`?

**Comment by styfle** [comment]
Fixed in #378
